### ISSUE TITLE
Disjunction/patterns for files in NugetPackager

### ISSUE THREAD

**Issue by apskim** [Microsoft/vso-agent-tasks#626]
The NugetPackager task claims to support minimatch syntax in its "Pattern" field, but it actually doesn't. Since there's no documentation for `Find-Files`, we spent several hours trying to understand which pattern features it supports at all.

Our scenario is simple: within a certain folder (let's even say in the entire repo), find all `*.csproj` files that are named in a certain way (for instance, `Contoso.{Core,Utils,Frontend}.csproj`, and package them). However, all ways to provide some form of disjunction seem to be unsupported by NugetPackager. That includes:
* Brace expansion: `Contoso.{Core,Utils,Frontend}.csproj`
* Extended glob patterns:  `Contoso.@(Core|Utils|Frontend).csproj`
* Space-separated list: `Contoso.Core.csproj Contoso.Utils.csproj Contoso.Frontend.csproj`

We ended up creating 30+ individual packager tasks in our build configuration, one per package. That's incredibly ugly, so I really wish NugetPackager supported at least some form of disjunction in its pattern language other than `*`, `?`, and `**` from good old DOS times.

**Comment by pascalberger** [comment]
This seems to not only affect NugetPackager, but also quite a lot of other tasks (which also claim to support minimatch): https://github.com/Microsoft/vso-agent-tasks/search?p=3&q=find-files&utf8=%E2%9C%93

It would be really helpful if the PowerShell modules are documented somewhere. I asked for this a while ago in #352, but don't know about its state since its now tracked internally.

**Comment by ericsciple** [comment]
Thanks! I checked the tasks. NuGet Packager is the only one that claims to support minimatch other than Publish Artifacts. I've created a pull request to fix the description. I'll close out this issue after PR is merged.

Other tasks do use the Find-Files cmdlet which performs pattern matching (does not claim to support minimatch).

The PowerShell modules are moving to GitHub soon. I'm hoping I'll be able to start on that in the next couple weeks.
